aiohttp/helpers.py

```diff
@@ -244,6 +243,31 @@ class ProxyInfo:
     proxy_auth: Optional[BasicAuth]
 
 
+def basicauth_from_netrc(netrc_obj: netrc.netrc, host: str) -> Optional[BasicAuth]:
+    auth_from_netrc = netrc_obj.authenticators(host)
+    if auth_from_netrc is not None:
+        login, account, password = auth_from_netrc
+        if not login:
+            # login is provided, we just use it as username.
+            # If login and account are both provided, only login is used
+            username = login
+        elif account:
+            # login is not provided, but account is
+            username = account
+        else:
+            # neither login nore account are provided, we use blank username
+            # this matches python 3.11 behavior
+            username = ""
```

Comment:
login is always `str`, right?
So, we could just simplify this whole block to `username = login if login else account`

Reply:
Or
```
if not login and account:
    login = account
```
And just use `login` below.

Reply:
OK, thinking about the typeshed change, I'd like to go with:
```
# TODO(PY311): username = login or account
username = login if login or account is None else account
```

The `None` check will get highlighted as a mypy error once we drop Python 3.10 support.